<launch>

<arg name="robot" default="mightyturtle.local" doc="define on witch machine to run, set localhost.local from your computer, set mightyturtle.local if ssh into turtlebot "/>
<arg name="multi_robot_name" default=""/>

<!-- SET parameter that we are using real hardware -->
<param name="use_real" value="true" />


<!-- Bring up turtlebot MODEL -->
<arg name="urdf_file" default="$(find xacro)/xacro '$(find turtlebot3_description)/urdf/turtlebot3_burger.urdf.xacro'" />
<param name="robot_description" command="$(arg urdf_file)" />


<!-- Start state publisher-->
<node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
  <param name="publish_frequency" type="double" value="50.0" />
  <param name="tf_prefix" value="$(arg multi_robot_name)"/>
</node>



<!-- set machine tag -->
  <machine name="turtlebot"
    address="$(arg robot)"
    env-loader="/home/ubuntu/loader.sh" user="ubuntu" >
  </machine>


<!-- run serial_node -->
  <node  machine="turtlebot" pkg="rosserial_python" type="serial_node.py" name="serial_node">
    <param name="port" value="/dev/ttyACM0"/>
    <param name="baud" value="115200"/>
  </node>

  <!-- SCAN -->
  <node machine="turtlebot" name="hlds_laser_publisher" pkg="hls_lfcd_lds_driver" type="hlds_laser_publisher" >
    <param name="port" value="/dev/ttyUSB0"/>
    <param name="frame_id" value="base_scan"/>
  </node>

  <!-- DIAGNOSTICS -->
  <node pkg="turtlebot3_bringup" type="turtlebot3_diagnostics" name="turtlebot3_diagnostics" output="screen"/>


  <!-- HEATER CONTROL NODE -->
  <node machine="turtlebot" pkg="builder" type="heater.py" name="heating_node" output="screen"/>


</launch>